File: XCode iOS Project/Old/VisualNovelReader/VisualNovelReader/StoryViewController.m
//
//  StoryViewController.m
//  Visual Novel Reader
//
//  Created by Christopher Raleigh on 2015/16/06.
//  Copyright (c) 2015 Christopher Raleigh. All rights reserved.
//

#import "StoryViewController.h"

@implementation StoryViewController {
    IBOutlet UINavigationController <PReaderNavigationController, PStoryReader, PStoryNavigator> * navigationController;
    StoryView * storyView;
    Bookmark * bookmark;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    storyView = [[StoryView alloc] initWithNavigationController:navigationController];
    [[self view] addSubview:storyView];
    bookmark = [[Bookmark alloc] initWithStoryView:storyView andController:navigationController];
}

- (void)viewWillAppear:(BOOL)animated {
    [navigationController setNavigationBarHidden:true animated:animated];
    [super viewWillAppear:animated];
}

- (void)viewWillDisappear:(BOOL)animated {
    [navigationController setNavigationBarHidden:false animated:animated];
    [super viewWillDisappear:animated];
}

- (void) showPages : (NSArray *) pages {
    [bookmark showPages:pages];
}

- (void) onButtonContinue {
    [bookmark onButtonContinue];
}

- (void) onButtonPrevious {
    [bookmark onButtonPrevious];
}

- (void) onButtonNext {
    [bookmark onButtonNext];
}

@end

File: XCode iOS Project/Phantom of the West/Phantom of the West/StoryViewController.swift
//
//  StoryViewController.swift
//  Phantom of the West
//
//  Created by Christopher Raleigh on 2015/18/07.
//  Copyright (c) 2015 Christopher Raleigh. All rights reserved.
//

import UIKit

/*
Initializes with a view controller and story.
The view controller pops and pushes the options menu as needed.
The story implements the PStory protocol.
*/
public class StoryViewController: UIViewController, PStoryObserver {
    
    /*
    Set an event handler so that the navigation buttons react.
    */
    var storyViewHandler: PNavigatorHandler? {
        get {
            return storyView.handler
        }
        set {
            storyView.handler = newValue
        }
    }
    
    /*
    Set to true if the user should be shown iAds.
    For example, the user might not have bought advertisment removal.
    */
    /*var shouldDisplayAdvertisement: Bool {
        get {
            return adHandler.shouldDisplayAdvertisement
        }
        set {
            adHandler.shouldDisplayAdvertisement = newValue
        }
    }*/
    
    private lazy var storyView: StoryView = StoryView()
    
    /*private var adHandler: Advertising = Advertising()*/
    
    private lazy var bookmark: Bookmark = Bookmark(storyView: self.storyView)
    
    override public func viewDidLoad() {
        super.viewDidLoad()
        storyView.view = view
    }
    
    override public func viewWillTransitionToSize(size: CGSize,
        withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
            super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
            storyView.removeOrientationConstraints()
            coordinator.animateAlongsideTransition(nil, completion: {
                _ in
                self.storyView.addOrientationConstraints(size)
            })
    }
    
    public override func viewWillAppear(animated: Bool) {
        if let nc = navigationController {
            nc.navigationBarHidden = true
        }
        
        // Enable advertising if ad-removal not purchased.
        canDisplayBannerAds = !SaveManager.purchasedAdRemoval
        
        super.viewWillAppear(animated)
    }
    
    public override func viewWillDisappear(animated: Bool) {
        if let nc = navigationController {
            nc.navigationBarHidden = false
        }
        super.viewWillDisappear(animated)
    }
    
    public func update(pages: [PStoryPage]) {
        bookmark.update(pages)
    }
    
    /*
    The continue button is pressed.
    */
    func onButtonContinue() {
        bookmark.onButtonContinue()
    }
    
    /*
    The previous button is pressed.
    */
    func onButtonPrevious() {
        bookmark.onButtonPrevious()
    }
    
    /*
    The next button is pressed.
    */
    func onButtonNext() {
        bookmark.onButtonNext()
    }
}